[CS2113T-T09-4] easyLog

ongweisheng opened the pull request:
easyLog is a command line interface (CLI) application for warehouse employees to manage their items and orders in their warehouse. It is optimized for CLI users so that frequent tasks can be done faster by typing in commands.

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+## Table of Contents
+
+- [Introduction](#introduction)
+    * [To be added](#to-be-added)
+        + [To be added](#to-be-added)
```
Table of contents is incomplete

tzexern replied:
> Is it better to include the links to each section right after it is created so that it won't be missed out in the future?

I agree with `kewenlok`.
Maybe it is better to do so as the table of contents may grow larger than expected?

kewenlok commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+## Table of Contents
+
+- [Introduction](#introduction)
+    * [To be added](#to-be-added)
+        + [To be added](#to-be-added)
```
Is it better to include the links to each section right after it is created so that it won't be missed out in the future?

oscarlai1998 replied:
Links are broken and not working

LJ-37 replied:
Table of contents incomplete.

fupernova replied:
Table of contents is incomplete.

ivancze commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+The EasyLog component initializes the app and looks for any available save file to load. It then proceeds to
+continuously take in input from the user until the user types in a exit command.
+
+### 2.3 Storage Component
+
+![Storage Diagram](https://user-images.githubusercontent.com/57165946/112991416-ab022380-9199-11eb-9ead-cdfcd19cf8e5.png)
+
```
I think the use of the rectangle with the clipped corner to label your diagram is incorrect? I think that label is mainly used for labelling for loops, while loops etc. in sequence diagrams.

This usage is seen in other diagrams as well.

![rectangle clipped](https://user-images.githubusercontent.com/70261399/113076429-9a859380-9201-11eb-9d2c-bd9a543a9907.JPG)

oscarlai1998 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+1. Fork the easyLog repository [here](https://github.com/AY2021S2-CS2113T-T09-4/tp) and git clone it to any location on
+   your computer.
+
```
There is no subsection on how to read the developer guide.

ivancze commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+The ui component deals with interactions with the user by displaying the appropriate messages according to the users
+input.
+
+### 2.5 Parser Component
+
+![Parser Diagram](https://user-images.githubusercontent.com/57165946/112991918-42677680-919a-11eb-8868-79391cfc2c9e.png)
+
```
Perhaps consider using a caption at the bottom of the diagram to label it instead?

fupernova commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+### 2.3 Storage Component
+
+![Storage Diagram](https://user-images.githubusercontent.com/57165946/112991416-ab022380-9199-11eb-9ead-cdfcd19cf8e5.png)
```
For a class diagram, "( )" should be appended to class methods to ensure clear distinction between class variables and class methods. Also, all class variables should be grouped under one compartment in the class and all class methods should be grouped another compartment of the class instead of each attribute and method having its own separate compartment, like in the class Receipt.
![112991416-ab022380-9199-11eb-9ead-cdfcd19cf8e5](https://user-images.githubusercontent.com/60054901/113076666-226b9d80-9202-11eb-8bd0-212badc1e337.png)

kewenlok replied:
The UML class diagram shown below for the storage component seems wrong. The SaveData and Receipt class should only have 3-row boxes, one for the class name, one for attributes and the last one for method. The methods should be combined in one box.

![image](https://user-images.githubusercontent.com/70790702/113076443-a40efb80-9201-11eb-9ac3-918b42e90c96.png)

blasco-git replied:
![image](https://user-images.githubusercontent.com/66361650/113076797-6c548380-9202-11eb-9bb6-667b3b285157.png)
It would have been better to include an empty attribute box for SaveData.
Also, the methods should contain a pair of brackets at the end

LJ-37 replied:
methods should have a () after the method name to show difference between class methods and class variables. Methods and variables should be grouped together respectively in 1 compartment in the diagram each.
![image](https://user-images.githubusercontent.com/71820915/113077772-5e076700-9204-11eb-8792-9da235d63a47.png)

marklowsk replied:
I am not sure if you are missing the method `()` brackets or you did not follow the standard for UML class diagrams.
`attribute types` are also missing.

tzexern replied:
![image](https://user-images.githubusercontent.com/65816506/113094989-44c2e280-9225-11eb-80d5-607c3a185e17.png)

Should the top left box be omitted instead? (This can be seen in other UML diagrams in this DG as well)

For the Receipt class, should there be three rows instead? (Similar occurrences can be seen in other UML diagrams in this DG as well)

oscarlai1998 commented on docs/DeveloperGuide.md:
````diff
@@ -1,34 +1,178 @@
+
+5. Upon successful run, the following opening message will be shown:
+
+```
````
There are no instructions on how to set up with the jar file. This should be added so users know how to run using the jar file

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+### 2.6 Command Component
+
+![Command Diagram](https://user-images.githubusercontent.com/57165946/112992782-15679380-919b-11eb-8c46-436ba701a754.png)
```
Diagram is inaccurate, the use of arrows to represent parent and child classes are wrong. Consider using a triangle for an arrow head (the arrow head is a bit ambiguous) or follow the previous UML diagrams for consistency. Can also consider simplifying the diagram by omitting the irrelevant or less important details.

marklowsk replied:
![image](https://user-images.githubusercontent.com/71376412/113076825-78d8dc00-9202-11eb-81e9-d3bf66f2ff32.png)
I think there are some mistakes in the diagram:
`{abstract}` keyword not stated in diagram.
`extends` not used in diagram.

kewenlok replied:
The command component UML diagram seems a bit cluttered and too complex. The arrow in the UML class diagram should be pointing to the parent class that the class is inherited from instead of the other way round. You may consider extract out the child of OrdersCommand and ItemsCommand into separate diagrams to provide multi-level views without being too complex.

![image](https://user-images.githubusercontent.com/70790702/113077610-0c5edc80-9204-11eb-8ae2-d60060d425d6.png)

blasco-git replied:
![image](https://user-images.githubusercontent.com/66361650/113077654-1f71ac80-9204-11eb-88ec-99b3908543f6.png)
![image](https://user-images.githubusercontent.com/66361650/113077668-28fb1480-9204-11eb-8f39-3d89f5ba06e6.png)
repeatedOrdersAdd is missing its parameters

fupernova replied:
The arrows from `Command` to its 4 subclasses should be the other way round. Also, hollow arrowheads should be used instead of solid arrowheads to represent inheritance, like what yall did for the Parser and Storage.

Please ensure that it is reflected in your diagram that `Command` is an abstract class by including the tag `{abstract}`.
![112992782-15679380-919b-11eb-8c46-436ba701a754](https://user-images.githubusercontent.com/60054901/113077939-bd657700-9204-11eb-95c1-20cb9d0c6cca.png)

LJ-37 replied:
Arrows should be pointing towards the parent class instead of the child. Diagram is too cluttered with confusing arrows and lines. Lines that intersect can be confusing as to where are the ends of the arrow.
![image](https://user-images.githubusercontent.com/71820915/113078451-b4c17080-9205-11eb-9ea3-f153f88e5f81.png)

LJ-37 replied:
Style of diagram significantly different from the previous 2 class diagrams. Can consider changing to the same format and style for better consistency throughout the developer guide.

ivancze commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+![Command Diagram](https://user-images.githubusercontent.com/57165946/112992782-15679380-919b-11eb-8c46-436ba701a754.png)
+
+Different Commands execute by the program.
```
I think there's a typo here (executed*) ?

oscarlai1998 replied:
I believed it should be executed instead of execute

fupernova commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing} <br> <br>
+
+[Return to Top](#introduction)
```
Please make sure to check that all links in your guide work.

ivancze replied:
Nice return to top feature! Maybe you have not implemented it so it does not work on the website. Otherwise, nice feature to have.

xseh replied:
Its is a good idea to put a link back to the introduction. Unfortunately, the link is broken and does not work.

LJ-37 replied:
Return to top link is broken

kewenlok commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+|v1.0|user|add new orders|track the orders added to the system|
+|v1.0|user|view the orders in the system|track the orders that are added or not added to the system|
+|v1.0|user|remove orders|remove the orders that are no longer required to be tracked|
+|v1.0|user|clear all items|re-track all the items if there are any problems|
+|v1.0|user|clear all orders|re-track all the orders if there are any problems|
+|v1.0|user|view all available commands|refer to them when I forget how to use the application|
+|v1.0|user|view all items-related commands|refer to them when I forget how to use the items feature|
+|v1.0|user|view all orders-related commands|refer to them when I forget how to use the orders feature|
+|v2.0|user|create, save data and load existing data|work on another device with the saved data|
+|v2.0|expert user|create, save data and load existing data|edit the data file directly|
+|v2.0|user|delete multiple items|save time by not deleting items one by one|
+|v2.0|user|delete multiple orders|save time by not deleting orders one by one|
+|v2.0|user|find relevant items|save time by looking through relevant items|
+|v2.0|user|find relevant orders|save time by looking through relevant orders|
+|v2.0|user|add a new item with a specified unit price|calculate order price more easily|
+|v2.0|user|update the unit price of an item|revise unit price or correct mistaken unit price entered earlier|
+|v2.0|user|add a new item with its stock|keep track of total number of stock and calculate order price more easily|
+|v2.0|user|update the stock of an item|Correct mistaken stock number entered earlier|
+|v2.0|user|specify the quantity of a item to add to the order|calculate the total price of the order and update the item inventory correctly.|
+|v2.0|user|check the total price of a specific order|feedback to customer when they ask for it and do not have to look through the order list.|
+|v2.0|user|update the status of an order|differentiate the orders by their status.|
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+- Be able to work on any mainstream OS as long as it has Java 11 or above installed.
+- Be able to respond to any command in less than 2 seconds.
+- A user should be able to complete majority of tasks faster using CLI than GUI.
 
 ## Glossary
 
-* *glossary item* - Definition
+Abbreviation | Full title | Definition
+-------- | ---------- | ----------
+**CI**   | Continuous Integration | Combining parts of a software product to form a whole
+**IntelliJ** | IntelliJ | An Integrated Development Environment written in Java
+**CLI** | Command Line Interface | A program that accepts text inputs to execute operating system functions
+**GUI** | Graphical User Interface | An interface that allows users to interact through graphical icons
+**Mainstream OS** | Windows, Linux, Unix, OS-X | Operating systems
+
```
I like the way your team added technical jargon to the glossary. It makes things clearer for developers not familiar with certain terms used in the document.

fupernova replied:
I like the inclusion of a glossary of technical terms to help the people who are unfamiliar with these terms.

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+Different Commands execute by the program.
+
+### 2.7 Model Component
```
Empty. Missing description, UML diagrams and  other components.

oscarlai1998 reviewed (commented):
Generally the developer guide is missing a lot of information but the description used is pretty detailed and concise.

ivancze commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+## 2. Design
+
+### 2.1 Architecture
+
+![Architecture Diagram](https://user-images.githubusercontent.com/57165946/112990632-ddf7e780-9198-11eb-99ec-f29aafbfc04f.png)
+
+
```
Consider using the same line style throughout the diagram (and the entire DG as well)? Currently, three different line styles are used, straight angled lines, curved lines, and 90 degree lines (refer image)

Also consider rearranging the components to try avoid crossing over of lines?

![linestyle](https://user-images.githubusercontent.com/70261399/113077031-e38a1780-9202-11eb-825e-55ce9c29b1d0.JPG)

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+### 2.7 Model Component
+
+### 2.8 Exceptions Component
+Possible exceptions existing in the program.
```
Missing information. Details such as description, diagrams and other components can be added to make the component easier to understand for developers.

fupernova commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+### 1.2 Prerequisites
+
+- Java 11 (can be downloaded from [here](https://www.oracle.com/sg/java/technologies/javase-jdk11-downloads.html))
+- IntelliJ Idea (can be downloaded from [here](https://www.jetbrains.com/idea/download/#section=mac))
```
I like that you included links to the prerequisite software. Really helps someone new who is not tech savvy to figure out how to set your app up.

xseh replied:
Can consider adding information on the reasons for using this software and add indication of recommendation. For example, "Tip: IntelliJ is recommended as the DG gives instructions on how to set up the application using IntelliJ". The same applied for Java 11 (can be because of compatibility, etc. )

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+- Constants: Fixed values used in the program
+- Messages: Output response from the program.
+
+## Implementation
```
Empty. Consider populating the page with detailed descriptions of each class and other implementation details.

LJ-37 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+### 2.1 Architecture
+
+![Architecture Diagram](https://user-images.githubusercontent.com/57165946/112990632-ddf7e780-9198-11eb-99ec-f29aafbfc04f.png)
```
Such lines can be ambiguous as to where the line starts from, it could be from storage or command.
![DGreview1](https://user-images.githubusercontent.com/71820915/113077410-affbbd00-9203-11eb-88ae-49b318d07959.JPG)

marklowsk replied:
![image](https://user-images.githubusercontent.com/71376412/113077714-4203c580-9204-11eb-804b-e32d7ed138ca.png)

I'm not sure if this will count as a bug, but the aforementioned components are not mentioned in the document or the architecture diagram.

xseh replied:
Non-standard usage of diagram symbols. I believe the shapes used for labeling the storage component is inaccurate; it is used in UML to indicate labels of various types of code instead of as a label for the name of component. Consider using a note or omitting it entirely to avoid confusion. Applies to the rest of the diagrams. 

Also consider using labeling or captions to describe the type of diagram drawn (Object, class or sequence diagrams), making understanding the diagram easier.

kewenlok replied:
Is it better to realign the components so that the arrow lines do not overlap with each other?

tzexern replied:
![image](https://user-images.githubusercontent.com/65816506/113094719-c8300400-9224-11eb-9318-4b8220e72959.png)

Maybe it's better to avoid crossing of edges between the classes?

marklowsk commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+### 2.5 Parser Component
+
+![Parser Diagram](https://user-images.githubusercontent.com/57165946/112991918-42677680-919a-11eb-8868-79391cfc2c9e.png)
```
![image](https://user-images.githubusercontent.com/71376412/113077287-73c85c80-9203-11eb-86a9-a6d6bb0489bb.png)

![image](https://user-images.githubusercontent.com/71376412/113078152-1cc38700-9205-11eb-982d-190f99bd372a.png)

I think you should adhere to the UML standards from the above SS of our course material.

blasco-git replied:
![image](https://user-images.githubusercontent.com/66361650/113077279-6dd27b80-9203-11eb-8451-6ba92a973967.png)
![image](https://user-images.githubusercontent.com/66361650/113077299-7925a700-9203-11eb-889a-986080bed872.png)

It would be better if the parameters for a method is included in the class diagram. Methods without parameters can also have an empty pair of brackets at the end.

LJ-37 replied:
Too many compartments in each component of the diagram. 1 component should only have 3 components, name, class variables and class methods. Class variables and methods should be grouped together in the same compartments.
![image](https://user-images.githubusercontent.com/71820915/113078074-fe5d8b80-9204-11eb-83c8-1672ca30d698.png)

kewenlok replied:
You may consider adding a footer or caption for all diagrams in the guide to explain what is the diagram showing.

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing} <br> <br>
```
Missing details on manual testing.

ivancze commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+component. Each component consists of various classes that work in tandem, to ensure it meets the purpose of our
+program.
+
+The purpose of this developer guide is to allow any interested contributors, who wish to develop this application
+further or just curious about the workings of this application. This would allow potential contributors to be able to
+dive right in to improving the code, performance, features or even adding new features much more easily due to the
+understanding of the structure of the codebase and implementation of existing features.
+
```
Good job, purpose is stated! However, you might want to check the comma in the first sentence.

ivancze reviewed (commented):
Overall looks good! Diagram styles seem consistent and clean, just a few diagram and grammatical issues to check on.

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+
+The Architecture Diagram shown above illustrates the high-level design of easyLog. We will now proceed to explain each
+component and their respective functionalities below.
```
Further elaborations on how each components interacts with one another would improve the documentation. Consider adding more details such expanding on the various components  and a brief description of each one.

blasco-git reviewed (commented):
Good attempt overall! Contains minor issues that can easily be rectified

xseh commented on docs/DeveloperGuide.md:
````diff
@@ -1,34 +1,178 @@
+Looking for save data.
+Save data not found.
+```
+
````
Consider using various icons to indicate different levels of elaboration. Collate the icons and its usages into a table for easy reference. The icons and purpose allows developers to understand how to use the DG and it improves organisation and overall readability of the DG.

fupernova commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+### 1.3 Setting up
+
+1. Fork the easyLog repository [here](https://github.com/AY2021S2-CS2113T-T09-4/tp) and git clone it to any location on
+   your computer.
+
+2. Open any IDE (IntelliJ Idea preferred as mentioned in the prerequisites) and click `Configure` -> `Project Defaults`
+   -> `Project Structure` -> `New`
+
+3. Next, go to `Import Project` and select the build.gradle file.
+
+4. After opening the project, go to `src` -> `main` -> `java` -> `seedu.easylog` -> EasyLog and right click on the
+   EasyLog class.
+
+5. Upon successful run, the following opening message will be shown:
```
The instructions are clear, comprehensive and easy for anyone to follow. Nice!

marklowsk reviewed (commented):
I think your DG is heavily still in WIP, keep up the hard work and progress!

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+- loading, saving and creating save data.
+- generation of receipts when order is done.
+
+### 2.4 Ui Component
+
+The ui component deals with interactions with the user by displaying the appropriate messages according to the users
+input.
```
Further elaboration needed, consider adding some diagrams and detailed descriptions on the innerworkings of the UI components.

xseh reviewed (commented):
Overall needs more details and labeling, generally good flow and organisation. Could be improved with usage of icons and other markup to make the DG easier to read and understand. Consider standardising and simplifying the UML diagrams and use accurate symbols to represent each component.

tzexern commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,178 @@
+
+![Command Diagram](https://user-images.githubusercontent.com/60378963/113083308-92802080-920e-11eb-8e4b-66bc7fdf79f2.png)
+
+Different Commands execute by the program.
```
Maybe a more elaborate description is needed here?

tzexern reviewed (commented):
Good job for some parts in the DG!

Even though it's largely incomplete, it's good to clear up some misconceptions of the UML diagram requirements set by the module. 
Changes are required for better overall readability.

e00426142 committed:
Update Order.java

ongweisheng committed:
Merge pull request #200 from e00426142/Yiwen-Update-Ui-javaDoc

Updated UI with no extra line spacing

ongweisheng committed:
Add app initialization implementation

- App initialization diagram
- Explanation for steps during app initialization

zikunz committed:
Fix Bug: Decimal not Accepted as Valid Item Unit Price

ongweisheng committed:
Merge pull request #202 from ongweisheng/app-initialization-diagram

Add app initialization implementation

ongweisheng committed:
Merge branch 'master' into ZhuZikun-BigDecimalForItemPrice

ongweisheng committed:
Merge pull request #203 from zikunz/ZhuZikun-BigDecimalForItemPrice

Fix Bug: Decimal not Accepted as Valid Item Unit Price

ongweisheng committed:
Fix minor bugs regarding storage component

ongweisheng committed:
Merge pull request #204 from ongweisheng/save-data-ui-fix

Fix minor bugs regarding storage component

e00426142 committed:
updated Command Component

ongweisheng committed:
Merge pull request #205 from e00426142/Yiwen-Update-DeveloperGuide

updated Command Component

ongweisheng committed:
Merge branch 'master' into Update-user-guide

Cocokkkk committed:
Orders feature: check error messages and fix trim() problem.

ongweisheng committed:
Merge pull request #207 from Cocokkkk/Orders-check-error-message

Orders feature: check error messages and fix trim() problem.

zikunz committed:
Fix Bug and Improvement for Item Items Add
The items add command of this version of easyLog is consistent with its orders add command. Specifically, the user needs to input item price and item command in 1 line. Also, the user will be prompted again and again if a pair of valid item price and stock is not given. The coding style is also improved.

zikunz committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-T09-4/tp into ZhuZikun-FixItemsAddEXceptionsProblems

Fix Merge Conflict

zikunz committed:
Comment a JUnit Test to Pass CheckStyle Check

zikunz committed:
Remove Unnecessary Code

zikunz committed:
Fix One Error Message Bug

ongweisheng committed:
Merge pull request #208 from zikunz/ZhuZikun-FixItemsAddEXceptionsProblems

Fix items add consistency issues and exceptions problems

Cocokkkk committed:
Update Ui and Messages for updated items add feature.
Fix partial mismatching expections.

ongweisheng committed:
Merge pull request #210 from Cocokkkk/Items-check-error-message

Items add feature

Cocokkkk committed:
Update items update feature

ongweisheng committed:
Merge pull request #255 from Cocokkkk/Update-itemsUpdate

Update items update feature

e00426142 committed:
Update UserGuide.md

andyrobert3 commented:
![image](https://user-images.githubusercontent.com/15076300/113505772-8af4aa80-9573-11eb-8ca6-d4d864368def.png)

Appreciate the short description for each component here, to give relevant context

andyrobert3 commented:
![image](https://user-images.githubusercontent.com/15076300/113507510-89c87b00-957d-11eb-8716-dfc845af36f3.png)

Do these Command components not have member variables?

andyrobert3 commented:
![image](https://user-images.githubusercontent.com/15076300/113507718-d496c280-957e-11eb-841c-14b40d3e4fb6.png)

Really like the clean diagram for the app initialization, with the different color labels
Keep up the diagram quality for the other implemented features

andyrobert3 commented:
Overall very well written and styled Developer Guide!
Just ensure that the generated architecture diagrams are manually checked by someone

zikunz committed:
Update UG: Better Phrasing, Add Tips (Tell Users about Valid Inputs)

ongweisheng committed:
Merge pull request #257 from e00426142/Yiwen-Updates-UserGuide

Update UserGuide.md

ongweisheng committed:
Merge branch 'master' into ZhuZikun-UserGuide

ongweisheng committed:
Merge pull request #258 from zikunz/ZhuZikun-UserGuide

Update UG: Better Phrasing, Add Tips (Tell Users about Valid Inputs)

ongweisheng committed:
Update orders add argument to show <customer_name>

ongweisheng committed:
Merge pull request #260 from ongweisheng/bug-fixing

Update orders add argument to show <customer_name>

ongweisheng committed:
Merge branch 'master' into bug-fixing

ongweisheng committed:
Fix minor typo

ongweisheng committed:
Merge pull request #261 from ongweisheng/bug-fixing

Fix User Guide FAQ Q4 typo

ongweisheng committed:
Merge branch 'master' into bug-fixing

ongweisheng committed:
Update orders add to explain what happens to stock count

ongweisheng committed:
Merge pull request #262 from ongweisheng/bug-fixing

Update orders add to explain what happens to stock count

Cocokkkk committed:
Improve the system message after adding new stocks or updating new item price and stock.

ongweisheng committed:
Merge pull request #263 from Cocokkkk/Fix-Bugs

Improve the system message

zikunz committed:
Add ! in UG

ongweisheng committed:
Merge pull request #264 from zikunz/ZhuZikun-UserGuide

Add ! in UG

ongweisheng committed:
Fix stock input can take in zero

ongweisheng committed:
Merge pull request #266 from ongweisheng/orders-add-bug-fix

Fix stock input can take in zero

ongweisheng committed:
Merge branch 'master' into orders-add-bug-fix

ongweisheng committed:
Add limit to length of customer name while adding order

ongweisheng committed:
Update UG for orders add part and fix typos

ongweisheng committed:
Merge pull request #267 from ongweisheng/orders-add-bug-fix

Add limit to length of customer name while adding order

ongweisheng committed:
Merge branch 'master' into orders-add-bug-fix

ongweisheng committed:
Update exception message to show correct format for input

ongweisheng committed:
Merge pull request #268 from ongweisheng/orders-add-bug-fix

Update exception message to show correct format for input

ongweisheng committed:
Update exception messages when error line occurred

ongweisheng committed:
Merge pull request #269 from ongweisheng/save-data-bug-fix

Update exception messages when error line occurred

ongweisheng committed:
Update Ui when receipt generated

ongweisheng committed:
Update UG for receipts feature

ongweisheng committed:
Merge pull request #270 from ongweisheng/receipt-bug-fix

Update Ui when receipt generated

ongweisheng committed:
Merge branch 'master' into orders-add-bug-fix

ongweisheng committed:
Update UG for orders add feature

ongweisheng committed:
Merge pull request #271 from ongweisheng/orders-add-bug-fix

Update orders add feature in UG to reflect that adding the same item twice does not combine the stock of the item but count it as 2 separate additions

ongweisheng committed:
Merge branch 'master' into save-data-bug-fix

ongweisheng committed:
Fix item name unable to save properly because of spaces

ongweisheng committed:
Fix JUnit test

ongweisheng committed:
Merge pull request #272 from ongweisheng/save-data-bug-fix

Fix item name unable to save properly because of spaces

Cocokkkk committed:
Update UserGuide: the description of help, items, orders, items add, items list and items update

ongweisheng committed:
Add loadFile Diagram

ongweisheng committed:
Add load save data implementation

ongweisheng committed:
Merge pull request #275 from ongweisheng/loadFile-diagram

Add loading of save data implementation in DG

zikunz committed:
Update UG: Change light bulbs and exclamation marks to notebooks

ongweisheng committed:
Merge pull request #276 from zikunz/ZhuZikun-UserGuide

Update UG: Change light bulbs and exclamation marks to notebooks

Cocokkkk committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-T09-4/tp into UserGuide-Update

Cocokkkk committed:
Modify the previous version and delete format and expected output for help / items / orders / items add / items list / items update

Cocokkkk committed:
Format relevant parts.

Cocokkkk committed:
Fix some errors.

ongweisheng committed:
Merge branch 'master' into loadFile-diagram

ongweisheng committed:
Merge pull request #274 from Cocokkkk/UserGuide-Update

Update UserGuide

ongweisheng committed:
Update loadfile diagram

ongweisheng committed:
Merge branch 'master' into loadFile-diagram

ongweisheng committed:
Update loadFile diagram

ongweisheng committed:
Merge remote-tracking branch 'origin/loadFile-diagram' into loadFile-diagram

ongweisheng committed:
Merge pull request #278 from ongweisheng/loadFile-diagram

Update loadFile diagram

zikunz committed:
Change Items Add: Allow Items of Same Names to be Added (and Tracked Differently)

zikunz committed:
Update UG: Notes Regarding Items Add and Orders Clear

ongweisheng committed:
Add delete command and save file diagram

ongweisheng committed:
Merge pull request #281 from zikunz/ZhuZikun-ItemsAddAllowSameName

Change Items Add: Allow Items of Same Names to be Added (and Tracked …

zikunz committed:
Adding JUnit Tests for Items Add, Items Clear and Items Update

zikunz committed:
Comment Some JUnit Test Code to Pass some Automatic Checks

zikunz committed:
Fix CheckStyle Issues

zikunz committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-T09-4/tp into ZhuZikun-JUnitTesting

zikunz committed:
Fix CheckStyle Issues and Change JUnit Tests for New Code

zikunz committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-T09-4/tp into ZhuZikun-UserGuide

zikunz committed:
Change "System" to "Inventory"

zikunz committed:
Update UG: Items Update Notes

zikunz committed:
Change Messages: Allow Users to Update with the Same Price / Stock

ongweisheng committed:
Add multiple diagrams

ongweisheng committed:
Merge branch 'master' into diagrams

ongweisheng committed:
Merge pull request #291 from zikunz/ZhuZikun-ItemsUpdateSamePriceOrStock

Change Messages: Allow Users to Update with the Same Price / Stock

ongweisheng committed:
Merge branch 'master' into ZhuZikun-JUnitTesting

ongweisheng committed:
Merge pull request #286 from zikunz/ZhuZikun-JUnitTesting

Add JUnit test code for Items Add, Items Update and Items Clear

ongweisheng committed:
Merge branch 'master' into diagrams

ongweisheng committed:
Merge pull request #293 from ongweisheng/diagrams

Add diagrams into docs folder

Cocokkkk committed:
Code modification.

ongweisheng committed:
Merge pull request #294 from Cocokkkk/remove-orders-delete-by-name

Code modification.

Cocokkkk committed:
UG: orders delete

ongweisheng committed:
Merge pull request #295 from Cocokkkk/UserGuide-Update

UG: orders delete

zikunz committed:
Update UG: Introduction & Items Add & Items Update & Items Clear...

jiangqx0225 committed:
update delete items by index

zikunz committed:
Update UG

jiangqx0225 committed:
update UG for delete items by index
update UG for delete items find

jiangqx0225 committed:
minor update UG for delete items by index

jiangqx0225 committed:
update readme for easylog

ongweisheng committed:
Merge pull request #296 from zikunz/ZhuZikun-UserGuide

Update UG: Please refer below for more information

ongweisheng committed:
Merge branch 'master' into Qixiong_delete_by_index

ongweisheng committed:
Merge pull request #298 from 951553394/Qixiong_delete_by_index

Qixiong delete by index

Cocokkkk committed:
Extract two methods: items add back and item sales decrease when deleteing an order

Cocokkkk committed:
Improve code quality

Cocokkkk committed:
improve code quality

ongweisheng committed:
Merge pull request #299 from Cocokkkk/improve-orders-delete

Extract two methods

jiangqx0225 committed:
update tests in DG
added Junit tests in constants

zikunz committed:
Update DG: Cosmetics, Documentation Testing and Dev Ops

jiangqx0225 committed:
minor update tests in DG

jiangqx0225 committed:
minor update tests in DG

jiangqx0225 committed:
minor update tests in DG

ongweisheng committed:
Merge branch 'master' into ZhuZikun-DeveloperGuide

ongweisheng committed:
Merge pull request #301 from zikunz/ZhuZikun-DeveloperGuide

Update DG: Cosmetics, Documentation Testing and Dev Ops

ongweisheng committed:
Merge branch 'master' into Qixiong_DG-testing

zikunz committed:
Fix Bug in Orders OrdersClear
This version of easyLog makes sure all items belonging to the orders to be cleared will be added back to the inventory / item list

ongweisheng committed:
Merge pull request #300 from 951553394/Qixiong_DG-testing

update tests in DG

zikunz committed:
Fix CheckStyle Issues

ongweisheng committed:
Merge branch 'master' into ZhuZikun-OrdersClearBugFixing

ongweisheng committed:
Merge pull request #303 from zikunz/ZhuZikun-OrdersClearBugFixing

Fix Bug in Orders OrdersClear

zikunz committed:
Update UG

ongweisheng committed:
Add project portfolio  file for ongweisheng

ongweisheng committed:
Merge branch 'master' into project-portfolio

ongweisheng committed:
Merge pull request #304 from ongweisheng/project-portfolio

Add project portfolio file for ongweisheng

ongweisheng committed:
Merge branch 'master' into update-developer-guide

ongweisheng committed:
Add sequence diagrams without comments and steps for now

Cocokkkk committed:
fix bugs for 6 feature: items add, items update, items stats, items clear, orders delete, orders clear.
Update userguide for those 6 features.

Cocokkkk committed:
Merge branch 'master' into fix-feature-bugs

e00426142 committed:
Update UserGuide.md

e00426142 committed:
Update DeveloperGuide.md

e00426142 committed:
Create QiuYiwen.md

jiangqx0225 committed:
update sequence diagram for orders delete

jiangqx0225 committed:
update manual testing in appendix E

ongweisheng committed:
Merge pull request #306 from Cocokkkk/fix-feature-bugs

fix bugs for 6 features

ongweisheng committed:
Merge branch 'master' into Yiwen-Updates-UserGuide

ongweisheng committed:
Merge pull request #307 from e00426142/Yiwen-Updates-UserGuide

Update UserGuide.md

ongweisheng committed:
Merge branch 'master' into Yiwen-Update-DeveloperGuide

ongweisheng committed:
Merge pull request #308 from e00426142/Yiwen-Update-DeveloperGuide

Update DeveloperGuide.md

ongweisheng committed:
Merge branch 'master' into Yiwen-PPP

ongweisheng committed:
Merge pull request #309 from e00426142/Yiwen-PPP

Create QiuYiwen.md

ongweisheng committed:
Merge branch 'master' into Qixiong_orders_delete_diagram

ongweisheng committed:
Merge pull request #310 from 951553394/Qixiong_orders_delete_diagram

update sequence diagram for orders delete

ongweisheng committed:
Merge branch 'master' into update-developer-guide

ongweisheng committed:
Fix bug where user can add orders when all items have stock 0

ongweisheng committed:
Update diagrams for DG

ongweisheng committed:
Update UG

Cocokkkk committed:
DG Diagram: items add

ongweisheng committed:
Remove unused code

ongweisheng committed:
Update Parser Diagram

ongweisheng committed:
Add global catcher for all exceptions

Cocokkkk committed:
Upload items add diagram

ongweisheng committed:
Add logging messages

ongweisheng committed:
Update UG

ongweisheng committed:
Update DG

ongweisheng committed:
Fix logging class bug

Cocokkkk committed:
Change items Add to 3.5.

jiangqx0225 committed:
update PPP for Qixiong

ongweisheng committed:
Update text-ui-test

ongweisheng committed:
Merge pull request #311 from Cocokkkk/ItemsAdd_DG

DG Diagram: items add

ongweisheng committed:
Merge branch 'master' into Qxiong_PPP

ongweisheng committed:
Update EXPECTED.TXT

ongweisheng committed:
Merge pull request #313 from 951553394/Qxiong_PPP

update PPP for Qixiong

zikunz committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-T09-4/tp

ongweisheng committed:
Merge branch 'master' into update-developer-guide

ongweisheng committed:
Remove logo to fix runtest.sh

ongweisheng committed:
Merge remote-tracking branch 'origin/update-developer-guide' into update-documentation

ongweisheng committed:
Merge pull request #312 from ongweisheng/update-developer-guide

Update documentation and code enhancements

Cocokkkk committed:
Li Kexuan PPP

Cocokkkk committed:
Remove a unused method for adding a repeat order.

Cocokkkk committed:
Merge branch 'master' into PPP

ongweisheng committed:
Merge pull request #314 from Cocokkkk/PPP

Update PPP

zikunz committed:
Update DG

zikunz committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-T09-4/tp into ZhuZikun-FinalFixBugs

Cocokkkk committed:
update PPP

Cocokkkk committed:
Merge pull request #316 from Cocokkkk/PPP

update PPP

e00426142 committed:
updated SS for UG

e00426142 committed:
Update QiuYiwen.md

ongweisheng committed:
Merge pull request #318 from e00426142/Yiwen-PPP

Update QiuYiwen.md

ongweisheng committed:
Merge branch 'master' into Yiwen-Updates-UserGuide

ongweisheng committed:
Merge pull request #317 from e00426142/Yiwen-Updates-UserGuide

updated SS for UG

ongweisheng committed:
Change EXPECTED.TXT

ongweisheng committed:
Merge pull request #319 from ongweisheng/fix-runtest.sh

Change EXPECTED.TXT

e00426142 committed:
Updated UG

ongweisheng committed:
Merge pull request #320 from e00426142/Yiwen-Updates-UserGuide

Updated UG

zikunz committed:
Final Changes: Test Code, UG, DG, PPP

ongweisheng committed:
Code cleanup

ongweisheng committed:
Fix checkstyle bug

ongweisheng committed:
Merge pull request #321 from ongweisheng/code-check-v2.1

Code cleanup

zikunz committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-T09-4/tp into ZhuZikun-FinalFixBugs

Resolve Merge Conflict

ongweisheng committed:
UG proofread

ongweisheng committed:
Merge pull request #323 from ongweisheng/ug-check-v2.1

UG proofread

zikunz committed:
Finalised UG

zikunz committed:
Finalised DG

jiangqx0225 committed:
Rename JiangQixiong.md to 951553394.md

ongweisheng committed:
Merge pull request #324 from AY2021S2-CS2113T-T09-4/951553394-patch-1

Rename JiangQixiong.md to 951553394.md

ongweisheng committed:
Merge branch 'master' into ZhuZikun-FinalFixBugs

zikunz committed:
Merge pull request #1 from ongweisheng/fix-zikun-bug

Please accept this PR

ongweisheng committed:
Merge pull request #322 from zikunz/ZhuZikun-FinalFixBugs

Zhu zikun final fix bugs

Cocokkkk committed:
Rename PPP

ongweisheng committed:
Proofread DG

ongweisheng committed:
Fix checkstyle bug

ongweisheng committed:
Fix checkstyle bug

ongweisheng committed:
Fix test errors

ongweisheng committed:
Merge pull request #325 from ongweisheng/dg-check-v2.1

Proofread DG

jiangqx0225 committed:
minor update PPP for Qixiong

ongweisheng committed:
PPP final update

Cocokkkk committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-T09-4/tp into PPP

ongweisheng committed:
Merge pull request #326 from 951553394/Qixiong_PPP_v1.1

minor update PPP for Qixiong

ongweisheng committed:
Merge branch 'master' into ppp-check-v2.1

Cocokkkk committed:
Merge branch 'master' into PPP

ongweisheng committed:
Merge pull request #327 from ongweisheng/ppp-check-v2.1

PPP final update

e00426142 committed:
update UG and PPP

ongweisheng committed:
Merge branch 'master' into PPP

ongweisheng committed:
Merge pull request #328 from Cocokkkk/PPP

Rename PPP

ongweisheng committed:
Merge branch 'master' into Yiwen-PPP

ongweisheng committed:
Merge pull request #329 from e00426142/Yiwen-PPP

update UG and PPP

ongweisheng committed:
Final check for submission

zikunz committed:
Update Zikun's PPP fast

ongweisheng committed:
Fix checkstyle bug

ongweisheng committed:
Merge pull request #330 from AY2021S2-CS2113T-T09-4/zikunz-patch-1

Update Zikun's PPP fast

ongweisheng committed:
Merge branch 'master' into final-check-v2.1

ongweisheng committed:
Edit PPP file name for zikun

ongweisheng committed:
Merge pull request #331 from ongweisheng/final-check-v2.1

Final check for submission

ongweisheng committed:
Update PPP links

ongweisheng committed:
Merge pull request #332 from ongweisheng/change-ppp-link

Update PPP links

jiangqx0225 committed:
minor update ug items delete

ongweisheng committed:
Merge pull request #333 from 951553394/qixiong_dg_update

minor update ug items delete

ongweisheng committed:
Merge branch 'master' into Update-user-guide

ongweisheng committed:
Fix documentation bug

ongweisheng committed:
Merge remote-tracking branch 'origin/Update-user-guide' into Update-user-guide

ongweisheng committed:
Merge pull request #334 from ongweisheng/Update-user-guide

Update documentation bug

e00426142 committed:
Update UserGuide.md

ongweisheng committed:
Merge pull request #335 from e00426142/Yiwen-Updates-UserGuide

Update UserGuide.md

zikunz committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-T09-4/tp

Resolve Merge Conflicts

zikunz committed:
Changes made for CS2101 UG

zikunz committed:
Do not touch others/ code

zikunz committed:
Do not touch others' code

zikunz committed:
Do not touch others' code

ongweisheng committed:
Merge pull request #337 from zikunz/ZhuZikun-CS2101UG

UG changes (CS2101)

zikunz committed:
Last changes for CS2101 UG

ongweisheng committed:
Merge pull request #338 from zikunz/ZhuZikun-CS2101UG

Last changes for CS2101 UG
